benchopt/base.py

```diff
@@ -227,11 +227,12 @@ def warm_up(self):
         ...
 
     def _warm_up(self):
-        if getattr(self, '_warmup_done', None):
+        cls = self.__class__
+        if getattr(cls, '_warmup_done', False):
             # already warmed up
             return
         self.warm_up()
-        self._warmup_done = True
+        cls._warmup_done = True
```

Comment:
I think in some cases, the warm_up depends on the parameters. So not sure this can be done like this.